src/mock-manager.ts

```diff
@@ -23,18 +23,23 @@ export class MockManager {
   /**
    * Mock a JSON-RPC request
    * @param method JSON-RPC method name
-   * @param data Data to be resolved, or error to be thrown in case of throw
+   * @param data Data to be resolved, or function to be called, or error to be thrown in case of throw
    * @param mockOptions Options of the mock
    * @example ```ts
    * // Mock one "eth_accounts" request
    * mockManager.mockRequest("eth_accounts", ["0x..."]);
    * // Persistently mock "eth_chainId" request
    * mockManager.mockRequest("eth_chainId", "0x1", { persistent: true });
+   * // Dynamicly mock "personal_sign" request
+   * mockManager.mockRequest("personal_sign", async (params: any) => {
+   *   let statement = (params as string[])[0];
+   *   return await bobsWallet.signMessage(statement);
+   * });
```

Comment:
Could you please also add this comment to the `mockRequest` of the `testing-utils.ts` file please? 🙏 

And since the feature you have done is actually super generic, we can be a bit more ambitious on the message in my opinion, what do you think about something like
```
 * // Mock with a dynamical value based on params
 * // "personal_sign" in this case
* mockManager.mockRequest("personal_sign", async (params: string[]) => {
*   const statement = params[0];
*   return await bobsWallet.signMessage(statement);
* });
```